include/criterion/assert.h

```diff
@@ -30,109 +35,1041 @@
+ * Passes if Actual is lexicographically greater than Reference.
+ * Otherwise the test is marked as failure but the execution will continue.
+ * The optional string is printed on failure.
+ *
+ * @param[in] Actual String to test
+ * @param[in] Reference Reference String
+ * @param[in] fmt printf like format string
+ *
+ *****************************************************************************/
+
 # define cr_expect_str_gt(...) CR_EXPAND(cr_assert_str_op_va_(CR_FAIL_CONTINUES_,  >, __VA_ARGS__))
 
+
+/**
+ * Passes if Actual is lexicographically greater or equal to Reference
+ *
+ * Call: \c cr_assert_str_geq(Actual, Reference, [FormatString, [Args...]])\n
+ * Passes if Actual is lexicographically greater or equal to Reference.
+ * Otherwise the test is marked as failure and the execution of the function
+ * is aborted.
+ * The optional string is printed on failure.
+ *
+ * @param[in] Actual String to test
+ * @param[in] Reference Reference String
+ * @param[in] fmt printf like format string
+ *
+ *****************************************************************************/
 # define cr_assert_str_geq(...) CR_EXPAND(cr_assert_str_op_va_(CR_FAIL_ABORT_,     >=, __VA_ARGS__))
+
+/**
+ * Passes if Actual is lexicographically greater or equal to Reference
+ *
+ * Call: \c cr_expect_str_geq(Actual, Reference, [FormatString, [Args...]])\n
+ * Passes if Actual is lexicographically greater or equal to Reference.
+ * Otherwise the test is marked as failure but the execution will continue.
+ * The optional string is printed on failure.
+ *
+ * @param[in] Actual String to test
+ * @param[in] Reference Reference String
+ * @param[in] fmt printf like format string
+ *
+ *****************************************************************************/
+
 # define cr_expect_str_geq(...) CR_EXPAND(cr_assert_str_op_va_(CR_FAIL_CONTINUES_, >=, __VA_ARGS__))
 
-// Array assertions
+/**@}*/
 
+/**
+ * @defgroup ArrayAsserts Array assertions
+ * @{
+ */
+
+/**
+ * Passes if Actual is byte-to-byte equal to Expected
+ *
+ * Call: \c cr_assert_arr_eq(Actual, Expected, [FormatString, [Args...]])\n
+ * Passes if Actual is byte-to-byte equal to Expected.
+ * Otherwise the test is marked as failure and the execution of the function
+ * is aborted.
+ * The optional string is printed on failure.
+ *
+ * @param[in] Actual Array to test
+ * @param[in] Expected Expected array
+ * @param[in] fmt printf like format string
+ *
+ *****************************************************************************/
 # define cr_assert_arr_eq(...) CR_EXPAND(cr_assert_mem_op_va_(CR_FAIL_ABORT_,      ==, __VA_ARGS__))
+
+/**
+ * Passes if Actual is byte-to-byte equal to Expected
+ *
+ * Call: \c cr_assert_arr_eq(Actual, Expected, [FormatString, [Args...]])\n
+ * Passes if Actual is byte-to-byte equal to Expected.
+ * Otherwise the test is marked as failure but the execution will continue.
+ * The optional string is printed on failure.
```

Comment:
I think the note from the original docs is missing, adding a warning like "This should not be used on arrays of struct, consider using cr_assert_arr_eq_cmp instead." could be nice.
(same thing for `cr_assert_arr_neq`)